### PR TITLE
Using AclCommand,avoid call the global method loadcache in SimpleAclAuthorizer

### PR DESCRIPTION
**Pull request by StevenLuMT**
In the class Named AclCommand,configure SimpleAclAuthorizer,but no need call loadCache.
If there are a lot of topics, loadCache will be very slow.
now we have 20,000 topics in a new cluster,all these topics need to be authed for each project group.No matter how to run AclCommand, it will be very slow.
this updating can improve the running time from minutes to less than one second

### CODE REVIEW

File: core/src/main/scala/kafka/admin/AclCommand.scala [hunk 1]
```diff
@@ -585,6 +588,12 @@ object AclCommand extends Logging {
       .describedAs("command-config")
       .ofType(classOf[String])
 
+    val loadAclCacheOpt = parser.accepts("load-acl-cache", "If you just run AclCommand,use this switch to avoid loading all ACL cache.")
```
Comment (ijuma) [inline core/src/main/scala/kafka/admin/AclCommand.scala]: Can we not only load the cache if needed without an additional CLI flag?

**Review by ijuma — COMMENTED** [review]
Thanks for the PR, one question below.

File: core/src/main/scala/kafka/admin/AclCommand.scala (hunk 1 above)
Comment (StevenLuMT) [reply core/src/main/scala/kafka/admin/AclCommand.scala]: > Can we not only load the cache if needed without an additional CLI flag?
yes, you can.the loadAclCacheOpt is default "true"

**Comment by omkreddy** [comment]
@StevenLuMT I will take a look soon. thanks.

**Comment by rajinisivaram** [comment]
@StevenLuMT Thanks for the PR, looks useful. Since this is adding a new config for AclAuthorizer and a new option for the tool, it will be good to have a small KIP describing the changes and motivation.

**Comment by StevenLuMT** [comment]
@rajinisivaram finished,thanks for review this KIP 
https://cwiki.apache.org/confluence/display/KAFKA/KIP-546%3A+Using+AclCommand%2Cavoid+call+the+global+method+loadcache+in+SimpleAclAuthorizer

**Comment by rajinisivaram** [comment]
@StevenLuMT Thanks for writing the KIP. Can you start a DISCUSS thread on the dev mailing list for the KIP? The KIP process is described here: https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals.

**Comment by StevenLuMT** [comment]
> @StevenLuMT Thanks for writing the KIP. Can you start a DISCUSS thread on the dev mailing list for the KIP? The KIP process is described here: https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals.

@rajinisivaram @omkreddy @ijuma 
Finished.Sorry,caused by some things, I reply to you so late.
JIRA's URL is  https://issues.apache.org/jira/browse/KAFKA-9424
KIP's URL is https://cwiki.apache.org/confluence/display/KAFKA/KIP-566%3A+Using+AclCommand%2Cavoid+call+the+global+method+loadcache+in+SimpleAclAuthorizer

File: core/src/main/scala/kafka/admin/AclCommand.scala (hunk 1 above)
Comment (tedyu) [inline core/src/main/scala/kafka/admin/AclCommand.scala]: use this switch -> turn off this switch

File: core/src/test/scala/unit/kafka/security/authorizer/AclCommandLoadCacheTest.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+class AclCommandLoadCacheTest extends ZooKeeperTestHarness {
+
+  private val testUsers1 = "User:test1"
+  private val testUsers2 = "User:test2"
+  private val topic = "test.topic1"
+  private val notLoadCache = "false"
+
+  /** every pull request run all test case,it may run some diff erro,we can improve this,I think*/
+  @Test
+  def notLoadCacheTest(): Unit = {
```
Comment (tedyu) [inline core/src/test/scala/unit/kafka/security/authorizer/AclCommandLoadCacheTest.scala]: Can you add some assertion in the tests ?

File: core/src/main/scala/kafka/admin/AclCommand.scala (hunk 1 above)
Comment (StevenLuMT) [reply core/src/main/scala/kafka/admin/AclCommand.scala]: thanks,updated.

File: core/src/test/scala/unit/kafka/security/authorizer/AclCommandLoadCacheTest.scala (hunk 2 above)
Comment (StevenLuMT) [reply core/src/test/scala/unit/kafka/security/authorizer/AclCommandLoadCacheTest.scala]: thanks,added,please help me review.

Comment (StevenLuMT) [reply core/src/test/scala/unit/kafka/security/authorizer/AclCommandLoadCacheTest.scala]: > Can you add some assertion in the tests ?

thanks,added,please help me review.

File: core/src/main/scala/kafka/admin/AclCommand.scala (hunk 1 above)
Comment (StevenLuMT) [reply core/src/main/scala/kafka/admin/AclCommand.scala]: > use this switch -> turn off this switch

thanks,updated.

File: core/src/test/scala/unit/kafka/security/authorizer/AclCommandLoadCacheTest.scala (hunk 2 above)
Comment (StevenLuMT) [reply core/src/test/scala/unit/kafka/security/authorizer/AclCommandLoadCacheTest.scala]: @tedyu 
thanks,updated,and conflicts are resolved
please help me review.

Comment (StevenLuMT) [reply core/src/test/scala/unit/kafka/security/authorizer/AclCommandLoadCacheTest.scala]: @tedyu 
thanks,updated,and conflicts are resolved
please help me review.

**Comment by rajinisivaram** [comment]
@StevenLuMT Have you started a discussion thread on the mailing list for the KIP related to this PR? Please update https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals page as well to make sure you get the next free number and the free number is updated.

**Comment by StevenLuMT** [comment]
> @StevenLuMT Have you started a discussion thread on the mailing list for the KIP related to this PR? Please update https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals page as well to make sure you get the next free number and the free number is updated.

@rajinisivaram @tedyu 
Sorry,I misunderstood. now I have created a discussion thread and updated the cwiki.

**Comment by mimaison** [comment]
`AclAuthorizer` is now being removed, so closing

**Pull request closed without merging**